keanu-python/tests/test_sampling.py

```diff
@@ -84,8 +101,25 @@ def test_iter_returns_same_result_as_sample(algo: str) -> None:
     samples_dataframe = pd.DataFrame()
     [samples_dataframe.append(pd.DataFrame(list(next_sample.items()))) for next_sample in islice(iter_samples, draws)]
 
-    for vertex_id in samples_dataframe:
-        np.testing.assert_almost_equal(samples_dataframe[vertex_id].mean(), np.average(samples[vertex_id]))
+    for vertex in samples_dataframe:
+        np.testing.assert_almost_equal(samples_dataframe[vertex].mean(), np.average(samples[vertex]))
+
+
+@pytest.mark.mpl_image_compare(filename='test_iter_with_live_plot.png')
+def test_iter_with_live_plot() -> Any:
+    with Model() as m:
+        m.exp = Exponential(np.ones((2, 2)))
+        m.exp.observe(np.array([[3., 2.], [0., 1.]]))
+        m.bernoulli = Bernoulli(m.exp)
+    net = m.to_bayes_net()
+
+    fig, ax = plt.subplots(1, 1, squeeze=False)
+    samples = generate_samples(net=net, sample_from=net.get_observed_vertices(), live_plot=True, refresh_every=5, ax=ax)
+
+    for sample in islice(samples, 5):
```

Comment:
could you parametrize this test on the number of steps that you take in the slice?

Reply:
`test_iter_with_live_plot.png` is actually identical to `test_sample_with_live_plot.png`, so you don't need both

Reply:
what do you want to test exactly?

Reply:
even just to test 2 different values (e.g. 1 and 5) would be nice. It's not a big deal though.